lib/tui/widget.ml

```diff
@@ -201,19 +203,21 @@ type selected_node =
   | Dir_selected of {
       prev_total : int;
       pos : int;
-      children : Fs.tree array;
+      children : Fs.tree array Lazy.t;
     }
 
 let next_level_to_doc selected_node =
   (* Get the next level files *)
   match selected_node with
   | File_selected file_contents ->
       File_contents (file_contents_to_doc ~file_contents)
-  (* No children of a directory without children *)
-  | Dir_selected { children = [||]; _ } -> Empty_directory
-  (* Non-empty array of children *)
-  | Dir_selected { prev_total; pos; children } ->
-      Directory_contents (children_to_doc ~prev_total ~pos children)
+  | Dir_selected { children; prev_total; pos } -> (
+      let children = Lazy.force children in
+      match children with
+      (* No children of a directory without children *)
+      | [||] -> Empty_directory
+      (* Non-empty array of children *)
+      | _ -> Directory_contents (children_to_doc ~prev_total ~pos children))
```

Comment:
I think here it's fine to keep `Lazy.force` because there's no elegant way to pattern-match on a lazy pattern inside record.